src/platform/packages/shared/react/use_observable/src/use_observable.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { useCallback, useEffect, useRef, useSyncExternalStore } from 'react';
+
+const isDev = process.env.NODE_ENV !== 'production';
+
+export interface ObservableLike<T> {
+  subscribe: (listener: (value: T) => void) => {
+    unsubscribe: () => void;
+  };
+}
+
+// RxJS BehaviorSubject fits this shape (getValue or value)
+export interface BehaviorSubjectLike<T> extends ObservableLike<T> {
+  getValue?: () => T;
+  value?: T;
+}
+
+/**
+ * This is a better version of react-use/lib/useObservable that:
+ *  - works smoother with BehaviorSubjects (no double render on mount) and no need for initialValue
+ *  - uses useSyncExternalStore under the hood
+ *  - warns if observable$ identity changes between renders
+ * @param observable$
+ */
+export function useObservable<T>(observable$: BehaviorSubjectLike<T>): T;
```

Comment:
This is not a requirement for the sidebar, but overall it could be a nice gradual improvement for Chrome and Kibana. Can be extracted into a separate PR